LSP-typescript.sublime-settings

```diff
@@ -9,6 +9,13 @@
 			"includeCompletionsForImportStatements": true,
 			"includeCompletionsForModuleExports": true,
 			"includeCompletionsWithInsertText": true,
+			"includeInlayEnumMemberValueHints": false,
+			"includeInlayFunctionLikeReturnTypeHints": false,
+			"includeInlayFunctionParameterTypeHints": false,
+			"includeInlayParameterNameHints": "none",
+			"includeInlayParameterNameHintsWhenArgumentMatchesName": false,
+			"includeInlayPropertyDeclarationTypeHints": false,
+			"includeInlayVariableTypeHints": false,
 		}
 	},
 	"settings": {
```

Comment:
At first I was thinking more to add one (or two if we want to control TS and JS separately) higher-level setting that controls whether type hints are enabled. And all of those above would be enabled by default. That would make it easier for the user to toggle it.

But then the existing settings are already in a similar vein to those inline hint settings so maybe just be consistent...

Though those still would need to be passed through `initializationOptions.preferences` so then you'd need some custom code to does that dynamically?

Reply:
So I guess the requirements are that we have separate settings for TS and JS.

Possibly way to achieve that is to have those settings in `settings` object and dynamically update `init_options` from `on_pre_start`. Though the `languageId` is not provided so it would be a bit hacky...

Reply:
Or the `InlayHintsListener` could have the logic to enable/disable itself.

Reply:
If someone for example disabled only enum and function return type inlay hints, there is no way on the client to filter out only those inlay hints. So trying to achieve that on the client seems not doable.

> It would have to be supported in theia server itself.

That sound good to me.
I think that would go with that route. I will try to make a PR for that, although based on what I have seen yesterday, the thiea repo is a bit unfamiliar for me... don't want to break something 🙂

Reply:
The easiest thing they I can do is to do nothing and wait till someone reports a problem, and then act. 

That is also an option. 🙂

Reply:
> I will try to make a PR for that

Looks like we would need to update TS preferences per-request like it does for formatting request already:

https://github.com/typescript-language-server/typescript-language-server/blob/49ddbeaa6a8af0822da57bad788505e1b9c12111/src/lsp-server.ts#L605-L605

Reply:
Also, theia doesn't support workspace/configuration right now. It maybe would be a good time to add support for it now. Otherwise we would have to pass all that configuration through `initializationOptions` but ideally not through `initializationOptions.preferences` since that is sent directly to typescript and I wouldn't want to include non-typescript  settings there.

Reply:
created a draft for this https://github.com/typescript-language-server/typescript-language-server/pull/265